Assets/Scripts/SS3D/Systems/Tile/BuildChecker.cs

```diff
@@ -0,0 +1,131 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace SS3D.Systems.Tile
+{
+    public static class BuildChecker
+    {
+        public static bool CanBuild(TileObject[] tileObjects, TileObjectSo tileObjectSo, bool replaceExisting)
+        {
+            bool canBuild = true;
+
+            // TileSystem tileSystem = SystemLocator.Get<TileSystem>();
+            TileLayer placedLayer = tileObjectSo.layer;
+
+            // Cannot build if the layer is already occupied. Skip if we replace the existing object
+            if (!replaceExisting)
+                canBuild &= tileObjects[(int)placedLayer].IsEmpty();
+
+            // Cannot build anything unless a plenum is placed
+            if (placedLayer != TileLayer.Plenum)
+                canBuild &= CanBuildOnPlenum(tileObjects[(int)TileLayer.Plenum], tileObjectSo);
+
+
+            switch (placedLayer)
+            {
+                case TileLayer.WallMountHigh when canBuild:
+                case TileLayer.WallMountLow when canBuild:
+                    {
+                        canBuild &= CanBuildWallAttachment(tileObjects[(int)TileLayer.Turf], tileObjectSo);
+                        break;
+                    }
+                // No furniture inside walls
+                case TileLayer.FurnitureBase:
+                case TileLayer.FurnitureTop:
+                    {
+                        canBuild &= !IsWall(tileObjects[(int)TileLayer.Turf]);
+                        break;
+                    }
+                // No walls on furniture
+                case TileLayer.Turf when tileObjectSo.genericType == TileObjectGenericType.Wall:
+                    {
+                        canBuild &= tileObjects[(int)TileLayer.FurnitureBase].IsEmpty() &&
+                        tileObjects[(int)TileLayer.FurnitureTop].IsEmpty();
+                        break;
+                    }
+            }
+
+            return canBuild;
+        }
+
+        public static bool CanBuildWallCollision(TileObjectSo tileObjectSo, Direction dir, PlacedTileObject[] adjacentObjects)
```

Comment:
why is this public and "CanBuildWallAttachment" is not ? Any particular reason ?

Reply:
CanBuildWallAttachment unfortunately has to be called by the TileMap in order to be provided with which tiles are adjacent. Not ideal, so let me know if you have a better option here.

Reply:
` 
public static bool CanBuild(TileObject[] tileObjects, TileObjectSo tileObjectSo, bool replaceExisting, Direction dir, PlacedTileObject[] adjacentObjects
){

\\ some stuff ...

return canBuild && CanBuildWallCollision(tileObjectSo, dir, adjacentObjects);
}

private  static bool CanBuildWallCollision(TileObjectSo tileObjectSo, Direction dir, PlacedTileObject[] adjacentObjects)
        {
            // some stuff ..
        }
`

Am I missing something ?

Reply:
Resolved as suggested by making CanBuildWallCollision private and passing two additional parameters with CanBuild